### PR TITLE
Feature/706 ocpp 201 california pricing requirements

### PR DESCRIPTION
**Pull request by maaikez**
## Describe your changes
OCPP 201 california pricing requirements. This includes display message and tariffandcost

## Issue ticket number and link
#706 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] If OCPP 2.0.1: I have updated the [OCPP 2.0.1 status document](https://github.com/EVerest/libocpp/tree/main/doc/ocpp_201_status.md)
- [x] I read the [contribution documentation](https://github.com/EVerest/EVerest/blob/main/CONTRIBUTING.md) and made sure that my changes meet its requirements

### CODE REVIEW

**Comment by maaikez** [comment]
Made this a pull request now. I have to merge master still. And I have a question:

Should the metervalue triggers be stored to the database as well? So if libocpp crashes, the triggers will be stored?
It feels a bit too much but on the other hand, it is possible that it crashes and then no triggers will be sent anymore. Only the regular metervalues.

File: lib/ocpp/v201/charge_point.cpp [hunk 1]
```diff
@@ -631,6 +605,169 @@ void ChargePoint::configure_message_logging_format(const std::string& message_lo
...
+            DisplayMessage display_message;
+            display_message.message = message;
+            display_message.identifier_id = original_message.transactionInfo.transactionId;
+            display_message.identifier_type = IdentifierType::TransactionId;
+            this->callbacks.set_display_message_callback.value()({display_message});
+        }
+    }
+
+    // Check if cost is available and enabled, and if there is a totalcost message.
+    if (cost_enabled && response.totalCost.has_value() && this->callbacks.set_running_cost_callback.has_value()) {
```
Comment (hikinggrass) [inline lib/ocpp/v201/charge_point.cpp:636]: small nitpick: we typically use "and" "or" "not" instead of "&&" "||" "!"

**Review by hikinggrass — CHANGES_REQUESTED** [review]

File: include/ocpp/v201/charge_point_callbacks.hpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    /// \brief Callback function is called when the websocket connection status changes
+    std::optional<std::function<void(const bool is_connected)>> connection_state_changed_callback;
+
+    /// \brief Callback functions called for get / set / clear display messages
+    std::optional<std::function<std::vector<DisplayMessage>(const GetDisplayMessagesRequest& request /* TODO */)>>
+        get_display_message_callback;
+    std::optional<std::function<SetDisplayMessageResponse(const std::vector<DisplayMessage>& display_messages)>>
+        set_display_message_callback;
+    std::optional<std::function<ClearDisplayMessageResponse(const ClearDisplayMessageRequest& request /* TODO */)>>
+        clear_display_message_callback;
```
[the remark this answers is not in the document]
Comment (maaikez) [reply include/ocpp/v201/charge_point_callbacks.hpp:144]: no can be removed, thanks

File: lib/ocpp/v201/charge_point.cpp (hunk 1 above)
Comment (maaikez) [reply lib/ocpp/v201/charge_point.cpp]: ok done for all files i touched.

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**